alpha/apps/kaltura/lib/authenticationUtils.php

```diff
@@ -33,28 +33,60 @@ public static function generateNewSeed($userLoginData)
 	public static function addAuthMailJob($partner, $kuser, $mailType)
 	{
 		$loginData = $kuser->getLoginData();
+		$roleNames = $kuser->getUserRoleNames();
 		$loginData->setPasswordHashKey($loginData->newPassHashKey(kConf::get('user_login_qr_page_hash_key_validity')));
 		$loginData->save();
-		$resetPasswordLink = UserLoginDataPeer::getAuthInfoLink($loginData->getPasswordHashKey());
-		if(!$resetPasswordLink)
-		{
-			return null;
-		}
 		$partnerId = $partner->getId();
-		$publisherName = $partner->getName();
-		$bodyParams = array($kuser->getFullName(), $partnerId, $resetPasswordLink, $kuser->getEmail(), $partnerId, $publisherName, $publisherName, $kuser->getUserRoleNames(), $publisherName, $kuser->getPuserId());
+		
+		$roleNameToUseDynamicEmailTemplate = kEmails::getDynamicEmailUserRoleName($roleNames);
+		if ($roleNameToUseDynamicEmailTemplate)
+		{
```

Comment:
add all the logic here to a different function addDynamicContentAuthMailJob
so
if ($roleNameToUseDynamicEmailTemplate)
   return new function

then the old logic similar to what it was